### PR TITLE
Introduce Clay.Buffers for buffer & blit functions

### PR DESCRIPTION
**Pull request by grant-d**
Not sure where to put HashCode.Fnv - putting it in Clay.Buffers for now, else we can leave it in Clay but then we need a project (Nuget) dependency

### CODE REVIEW

File: src/SourceCode.Clay.Buffers/BufferSession.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        #region Constructors
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="BufferSession"/> struct.
+        /// </summary>
+        /// <param name="buffer">The buffer.</param>
+        /// <param name="result">The result.</param>
+        public BufferSession(byte[] buffer, ArraySegment<byte> result)
+        {
+            //Contract.Requires(buffer != null);
```
Comment (jcdickinson) [inline src/SourceCode.Clay.Buffers/BufferSession.cs]: Remove commented code. Use guard exceptions.

File: src/SourceCode.Clay.Buffers/BufferSession.cs [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        /// </summary>
+        public void Dispose()
+        {
+            if (Buffer == null)
+                return;
+
+            ArrayPool<byte>.Shared.Return(Buffer);
+            Buffer = null;
+        }
+
```
Comment (jcdickinson) [inline src/SourceCode.Clay.Buffers/BufferSession.cs]: Add implicit conversion operators.

Comment (grant-d) [inline src/SourceCode.Clay.Buffers/BufferSession.cs]: Per private chat, this needs significant design review but it will work for now.

**Comment by grant-d** [comment]
OK, fixed one CR, leaving second CR until later design review

**Pull request merged**